internal/maintenance/periodic_job_enqueuer.go

```diff
@@ -165,6 +168,12 @@ func (s *PeriodicJobEnqueuer) Start(ctx context.Context) error {
 				s.insertBatch(ctx, insertParamsMany)
 
 			case <-ctx.Done():
+				// Clean up timer resources. We know it has _not_ received from the
+				// timer since its last reset because that would have led us to the case
+				// above instead of here.
+				if !timerUntilNextRun.Stop() {
+					<-timerUntilNextRun.C
+				}
```

Comment:
I added this here to ensure we're stopping and draining the timer before returning from this loop.